Extension/src/LanguageServer/client.ts

```diff
@@ -2001,40 +2001,59 @@ export class DefaultClient implements Client {
     }
 
     private sendCustomBrowseConfiguration(config: any): Thenable<void> {
-        // config is marked as 'any' because it is untrusted data coming from a 3rd-party. We need to sanitize it before sending it to the language server.
-        if (!config || config instanceof Array) {
-            console.warn("discarding invalid WorkspaceBrowseConfiguration: " + config);
-            return Promise.resolve();
-        }
+        let lastCustomBrowseConfiguration: PersistentFolderState<WorkspaceBrowseConfiguration> = new PersistentFolderState<WorkspaceBrowseConfiguration>("CPP.lastCustomBrowseConfiguration", null, this.RootPath);
+
+        let sanitized: util.Mutable<WorkspaceBrowseConfiguration>;
+        while (true) {
```

Comment:
Just checking... This `while (true)` exists just so you can break out early when the config is set?  Can you add a brief comment to that effect?